### ISSUE TITLE
DoubleClick plugin: Dynamically change adTagUrl when another video in a playlist is played

### ISSUE THREAD

**Issue by linkurzweg** [kaltura/mwEmbed#4043]
Hello,

I want to change the adTagUrl for the Double Click plugin when another video in a playlist is requested, so that the correct video id and tags are sent when the video changes.

I have tried to update the adTagUrl  in the flashvars when the video changes, using setKPDAttribute like this:

` kdp.kBind("onChangeMedia.videoPlayerKaltura", function() {
                kdp.setKDPAttribute("configProxy.flashvars.doubleClick","adTagUrl", newAdTagUrl);
            });`

Unfortunately it doesn't work, the data in the request being sent to DoubleClick remains the same. What am I doing wrong?

**Comment by kaltura-hooks** [comment]
Hi @linkurzweg,

Thank for you reporting an issue and helping improve Kaltura!

To get the fastest response time, and help the maintainers review and test your reported issues or suggestions, please ensure that your issue includes the following (please comment with more info if you have not included all this info in your original issue):
 - Is the issue you're experiencing consistent and across platforms? or does it only happens on certain conditions?
    please provide as much details as possible.
 - Which Kaltura deployment you're using: Kaltura SaaS, or self-hosted?
	If self hosted, are you using the RPM or deb install?
 - Packages installed.
	When using RPM, paste the output for:
```
	# rpm -qa \"kaltura*\"
```	
	For deb based systems:
```	
	# dpkg -l \"kaltura-*\"
```
 - If running a self hosted ENV - provide the MySQL server version used
 - If running a self hosted ENV - is this a single all in 1 server or a cluster?
 - If running a self hosted ENV, while making the problematic request, run:
```
	# tail -f /opt/kaltura/log/*.log /opt/kaltura/log/batch/*.log | grep -A 1 -B 1 --color \"ERR:\|PHP\|trace\|CRIT\|\[error\]\"
```
   and paste the output.
 - When relevant, provide any screenshots or screen recordings showing the issue you're experiencing.

For general troubleshooting see:
https://github.com/kaltura/platform-install-packages/blob/Jupiter-10.13.0/doc/kaltura-packages-faq.md#troubleshooting-help

If you only have a general question rather than a bug report, please close this issue and post at:
http://forum.kaltura.org

Thank you in advance,

**Comment by linkurzweg** [comment]
OK, I got it to work now... The Attribute name is just "doubleClick", not "configProxy.flashvars.doubleClick". However, I still have the problem that I need to do the same with AMP... The AMP Kaltura component doesn't seem to be capable of this?

**Comment by linkurzweg** [comment]
Also, how can I read out the "AdKeys" of the played video? Unfortunately it can't be read from the playlist dataProvider :(

**Comment by OrenMe** [comment]
Hi @linkurzweg apologize for the delay in response.
Glad you have found your answer.
Are you changing the ad tag url cause you are sending entry related info on each change? You can use the evaluates param strings in your ad tag url and they will be evaluated on each media change, you can see a list of options here: http://player.kaltura.com/docs/api#evaluate

**Comment by linkurzweg** [comment]
@OrenMe Thank you very much! Yes, that is the exact reason why I am changing the ad tag url. I didn't know I could just use the evaluate param strings. This makes everything a lot easier :) I got it working fine but I still don't know how to access custom metadata correctly.
I have a field "AdKeys" in our custom metadata, I tried accessing it in the ad tag url like this:
`{mediaProxy.entryMetadata.AdKeys}`
That doesn't work unfortunately. Can you tell me how to do this correctly?

**Comment by OrenMe** [comment]
@linkurzweg can you share a test page?

**Comment by linkurzweg** [comment]
@OrenMe I can't publicly share pages on our development servers. but I can share a link from production: https://www.futurezone.de/digital-life/article226720201/Handy-bei-Gewitter-nutzen-Trifft-dich-jetzt-etwa-gleich-der-Blitz.html

If you scroll all the way down, there's a playlist player embedded into the page. I added some test data to the "Adkeys" property of the first video. Of course, this currently doesn't use the evaluate param strings in the ad tag url . But maybe it'll still help you in reconstructing the problem?  What's weird is, when I use `kdp.evaluate('{mediaProxy.entryMetadata}')`, it returns the object just fine, including the AdKeys property with  the text I entered. When I try `kdp.evaluate('{mediaProxy.entryMetadata.AdKeys}')` however, I always get `undefined` as a result.
